sys/core/src/TxsCore.hs

```diff
@@ -172,16 +173,25 @@ import VarId
 -- | TorXakis core main api -- start
 runTxsCore :: Config -> StateT s IOC.IOC a -> s -> IO ()
 runTxsCore initConfig ctrl s0  =  do
-     _ <- runStateT (runTxsCtrl ctrl s0)
-               IOC.EnvC { IOC.config = initConfig
-                         , IOC.unid   = 0
-                         , IOC.params = initParams
-                         , IOC.state  = initState
-                         }
-     return ()
-       where initState = IOC.Noning
-             initParams =
-               Map.union ParamCore.initParams SolveDefs.Params.initParams
+      _ <- runStateT (runTxsCtrl ctrl s0)
+              IOC.EnvC { IOC.config = initConfig
+                        , IOC.unid   = 0
+                        , IOC.params = updateParams initParams
+                                        $ Config.configuredParameters initConfig
+                        , IOC.state  = initState
+                        }
+      return ()
+      where initState = IOC.Noning
+            initParams =
+              Map.union ParamCore.initParams SolveDefs.Params.initParams
+
+updateParams :: ParamCore.Params -> [(Config.ParameterName,Config.ParameterValue)] -> ParamCore.Params
+updateParams = foldl applyConfParam
+  where applyConfParam allParams (Config.ParamName pnStr, Config.ParamValue pvStr) =
```

Comment:
Putting the type of the helper function might help understanding what's going on here:

```haskell
    applyConfParam :: ParamCore.Params -> (Config.ParameterName, Config.ParameterValue) -> ParamCore.Params
```

Reply:
Added